src/packagedcode/cargo.py

```diff
@@ -31,11 +106,14 @@ class CargoTomlHandler(models.DatafileHandler):
     @classmethod
     def parse(cls, location):
         package_data = toml.load(location, _dict=dict)
-
         core_package_data = package_data.get('package', {})
+        workspace = package_data.get('workspace', {})
 
         name = core_package_data.get('name')
         version = core_package_data.get('version')
+        if isinstance(version, dict) and "workspace" in version:
+            version = "workspace"
```

Comment:
is this really the case? workspace is used as version?

Reply:
Yes! See https://github.com/nexB/scancode-toolkit/blob/ff84e1cc04903c6bbb9d1eaae3794fe61a718594/tests/packagedcode/data/cargo/cargo_toml/tauri-examples/Cargo.toml 

And you're right this is weird to put this in the version (or other fields, as if they are not overwritten by the workspace values for some bug, they remian), using extra_data instead here.